core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -163,7 +164,11 @@ public void writePayload(InputStream message) {
     statsTraceCtx.outboundMessageSent(currentMessageSeqNo, currentMessageWireSize, written);
   }
 
-  private int writeUncompressed(InputStream message, int messageLength) throws IOException {
+  @VisibleForTesting
+  int writeUncompressed(InputStream message, int messageLength) throws IOException {
+    if (isClosed()) {
```

Comment:
Why are you making this change? I have already mentioned we have essentially this same thing in AbstractStream. What value does it add to have it here too?

You add it to `writeRaw()`, which is only called from `writeKnownLengthUncompressed()` which already has the isClosed() check. What is the objective?

Reply:
This was trying to avoid the allocation in https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/MessageFramer.java#L230 which is probably insufficient. I moved all the early exit logic into `writeRaw()` now.